### PR TITLE
EOS-12439: Implemented 8 unit tests for inputs.py classmethods.

### PR DESCRIPTION
**Pull request by sudarshana27**
Signed-off-by: Sudarshana Babar <sudarshana.babar@seagate.com>

### CODE REVIEW

File: test/api/python/provisioner/test_inputs.py [hunk 1]
```diff
@@ -744,3 +746,152 @@ def _check(*args):
...
+
+def test_ParserFiller_fill_parser_with_no_metadata_argparser():
+    parser = argparse.ArgumentParser()
+    SC = attr.make_class("SC", {
+        "y": attr.ib(
+            default=123,
+            type=int
+        )
+    })
+    ParserFiller.fill_parser(SC, parser)
```
Comment (andkononykhin2) [inline test/api/python/provisioner/test_inputs.py:826]: you don't do any checks here, what is the purpose of that test?

File: test/api/python/provisioner/test_inputs.py [hunk 2]
```diff
@@ -744,3 +746,152 @@ def _check(*args):
...
+
+
+def test_ParserFiller_fill_parser_input_checks(mocker):
+    parser = argparse.ArgumentParser()
+    SC = attr.make_class("SC", {
+        "y": attr.ib(
+            default=123,
```
Comment (andkononykhin2) [inline test/api/python/provisioner/test_inputs.py:833]: a boolean is expected here as a default value

File: test/api/python/provisioner/test_inputs.py [hunk 3]
```diff
@@ -744,3 +746,152 @@ def _check(*args):
...
+    })
+    ret = ParserFiller.extract_positional_args(SC, attr.fields_dict(SC))
+    assert ret == ([attr.fields(SC).x], {})
+
+
+def test_ParserFiller_extract_positional_args_no_metadata_argparser():
+    SC = attr.make_class("SC", {
+        "x": attr.ib(
+            type=int,
+            repr=False,
```
Comment (andkononykhin2) [inline test/api/python/provisioner/test_inputs.py:876]: why do we need `repr` here?

**Review by andkononykhin2 — COMMENTED** [review]
@sudarshana27 Looks good. Please just address minor questions. Thank you.

File: test/api/python/provisioner/test_inputs.py (hunk 1 above)
Comment (sudarshana27) [reply test/api/python/provisioner/test_inputs.py:826]: It was a suggestion given in the old PR : https://github.com/Seagate/cortx-prvsnr/pull/338/commits/f039adba116fb689a81dd706a99cd9f810a1d45c
ParserFiller is not covered but as long as it just passes result of the AttrParserArgs to argparse we should focus on special cases:

- a class has some argument that doesn't specify METADATA_ARGPARSER - that argument should be ignored, so calls to parser.parse_args should produce an error

This unit test is implemented to cover above test scenario, but it doesn't throw any error.
Please suggest further, whether should we keep it or remove ?

File: test/api/python/provisioner/test_inputs.py (hunk 3 above)
Comment (sudarshana27) [reply test/api/python/provisioner/test_inputs.py:876]: have added some random attributes to the variable as test scenario requirement is not to add metadata attr.

File: test/api/python/provisioner/test_inputs.py (hunk 2 above)
Comment (sudarshana27) [reply test/api/python/provisioner/test_inputs.py:833]: Done

File: test/api/python/provisioner/test_inputs.py (hunk 1 above)
Comment (andkononykhin2) [reply test/api/python/provisioner/test_inputs.py:826]: >so calls to parser.parse_args should produce an error
but you don't do that here

Comment (sudarshana27) [reply test/api/python/provisioner/test_inputs.py:826]: working on it

Comment (sudarshana27) [reply test/api/python/provisioner/test_inputs.py:826]: When I am calling parser.parse_args(), it is throwing below error:
`>           args = parser.parse_args(['--y', 'some-value'])

test/api/python/provisioner/test_inputs.py:828:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.7/argparse.py:1758: in parse_args
    self.error(msg % ' '.join(argv))
/usr/local/lib/python3.7/argparse.py:2508: in error
    self.exit(2, _('%(prog)s: error: %(message)s\n') % args)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ArgumentParser(prog='pytest', usage=None, description=None, formatter_class=<class 'argparse.HelpFormatter'>, conflict_handler='error', add_help=True)
status = 2, message = 'pytest: error: unrecognized arguments: --y some-value\n'

    def exit(self, status=0, message=None):
        if message:
            self._print_message(message, _sys.stderr)
>       _sys.exit(status)
E       SystemExit: 2

/usr/local/lib/python3.7/argparse.py:2495: SystemExit
`
It is not raising any exception as such in order to catch within pytest.raises.
Should I mark this test with pytest  marker as @pytest.mark.xfail ?

Comment (andkononykhin2) [reply test/api/python/provisioner/test_inputs.py:826]: @sudarshana27 yes, that is what we should expect, try the following:
```
    with pytest.raises(SystemExit) as excinfo:                                                                                                                               
       args = parser.parse_args(['--y', 'some-value'])
```
you may also check `unrecognized arguments: --y some-value'` in `excinfo`

Comment (sudarshana27) [reply test/api/python/provisioner/test_inputs.py:826]: Addressed

**Review by andkononykhin2 — APPROVED** [review]

**Pull request merged**